extensions/robot/driver/src/main/scala/org/virtuslab/ideprobe/robot/RobotProbeDriver.scala

```diff
@@ -0,0 +1,70 @@
+package org.virtuslab.ideprobe
+package robot
+
+import com.intellij.remoterobot.{RemoteRobot, SearchContext}
+import java.nio.file.Path
+import java.time.Duration
+import org.virtuslab.ideprobe.protocol.ProjectRef
+import org.virtuslab.ideprobe.robot.RobotSyntax._
+import scala.collection.mutable
+import scala.util.Try
+
+object RobotProbeDriver {
+  val robotPortProperty = "robot-server.port"
+
+  val pluginId = "com.jetbrains.test.robot-server-plugin"
+
+  private val robots: mutable.Map[Int, RemoteRobot] = mutable.Map.empty
+
+  def apply(driver: ProbeDriver): RobotProbeDriver =
+    driver.as(
+      pluginId, { driver =>
+        val port = getRobotPort(driver)
+        val robot = synchronized {
+          robots.getOrElseUpdate(port, new RemoteRobot(s"http://127.0.0.1:$port"))
+        }
+        new RobotProbeDriver(driver, robot)
+      }
+    )
+
+  private def getRobotPort(driver: ProbeDriver) = {
+    driver
+      .systemProperties()
+      .getOrElse(robotPortProperty, error(s"$robotPortProperty property not set in IntelliJ"))
+      .toInt
+  }
+}
+
+final class RobotProbeDriver(
+    driver: ProbeDriver,
+    val robot: RemoteRobot
+) extends SearchableComponent {
+
+  override protected def searchContext: SearchContext = robot
+  override protected def robotTimeout: Duration = RobotSyntax.robotTimeout
+
+  def openProject(path: Path): ProjectRef = {
+    val ref = driver.openProject(path)
+    closeTipOfTheDay()
+    checkBuildPanelErrors()
+    ref
+  }
+
+  def closeTipOfTheDay(): Unit = {
+    Try(robot.mainWindow.find(query.dialog("Tip of the Day")).button("Close").click())
```

Comment:
Is it intentional to catch everything here? ( know it was moved, so feel free to ignore)

Reply:
I is not needed for anything, I just close it because it is annoying and can cover something that would be important for screenshot, I don't want to fail the test if flaky robot fails.